test/integration/utils/environment.go

```diff
@@ -35,22 +40,31 @@ type TestBuild struct {
 	// TODO: Adding specific field for polling here, interval and timeout
 	// but I think we need a small refactoring to make them global for all
 	// tests under /test dir
-	Interval          time.Duration
-	TimeOut           time.Duration
-	KubeConfig        *rest.Config
-	Clientset         *kubernetes.Clientset
-	Namespace         string
-	StopBuildOperator chan struct{}
-	BuildClientSet    *buildClient.Clientset
-	PipelineClientSet *tektonClient.Clientset
-	Catalog           test.Catalog
+	Interval               time.Duration
+	TimeOut                time.Duration
+	KubeConfig             *rest.Config
+	Clientset              *kubernetes.Clientset
+	Namespace              string
+	StopBuildOperator      chan struct{}
+	BuildClientSet         *buildClient.Clientset
+	PipelineClientSet      *tektonClient.Clientset
+	Catalog                test.Catalog
+	Context                context.Context
+	BuildOperatorLogBuffer *bytes.Buffer
 }
 
 // NewTestBuild returns an initialized instance of TestBuild
 func NewTestBuild() (*TestBuild, error) {
 	namespaceID := gomegaConfig.GinkgoConfig.ParallelNode*200 + int(atomic.AddInt32(&namespaceCounter, 1))
 	testNamespace := "test-build-" + strconv.Itoa(int(namespaceID))
 
+	logBuffer := &bytes.Buffer{}
```

Comment:
I think there is no need to export operator-sdk here, I think we should do the following:

1. Keep this code without an import to sdk, but rather to our existing `ctxlog` pkg:
```go
	logBuffer := &bytes.Buffer{}

	l := ctxlog.NewLoggerTo(logBuffer, testNamespace)

	// l := zap.LoggerTo(logBuffer)
	// log.SetLogger(l)
	// l = l.WithName(testNamespace)

	ctx := ctxlog.NewParentContext(l)
```

2. Introduce a new function in the ctxlog package that can give you a logger, but that instead of sending it to STDOUT, it sends it to the buffer:
```go
// NewLoggerTo returns a new Logger which logs
// to a given destination.
func NewLoggerTo(destWriter io.Writer, name string) logr.Logger {
	l := zap.LoggerTo(destWriter)

	logf.SetLogger(l)

	l = l.WithName(name)

	return l
}

```

We will need to probably stop using operator-sdk import in the `ctxlog` in the future, depends on how much we need to refactor for the operator-sdk bump to v1.0, but for now I would like to keep integration tests without references to the operator-sdk pkg.

Reply:
Done.